app/src/main/java/org/fossasia/pslab/experimentsetup/electronicexperiments/DiodeExperiment.java

```diff
@@ -34,13 +34,17 @@ public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
             @Override
             public void onClick(View v) {
                 Intent intent = new Intent(getActivity(), OscilloscopeActivity.class);
-                if ("Half Wave Rectifier".equals(experiment))
-                    intent.putExtra("who", "Half Wave Rectifier");
-                else if ("Full Wave Rectifier".equals(experiment))
-                    intent.putExtra("who", "Full Wave Rectifier");
-                else
-                    intent.putExtra("who", "Diode Clipping Clamping");
-                startActivity(intent);
+                switch (experiment){
+                    case "Half Wave Rectifier" :
+                        intent.putExtra("who", "Half Wave Rectifier");
+                        break;
+                    case "Full Wave Rectifier" :
+                        intent.putExtra("who", "Full Wave Rectifier");
+                        break;
+                    default:intent.putExtra("who", "Diode Clipping Clamping");
+                             startActivity(intent);
+                             break;
```

Comment:
Should this break comes before `startActivity`?

Reply:
No according to the original code it should start the new intent when none of the other cases are applicable. I think there is no requirement of that break statement over there.